addons/test_mail/tests/test_mail_composer.py

```diff
@@ -656,10 +894,73 @@ def setUpClass(cls):
             'groups_id': [(4, cls.env.ref('base.group_partner_manager').id)],
         })
 
+
+    @users('employee')
+    @mute_logger('odoo.tests', 'odoo.addons.mail.models.mail_mail', 'odoo.models.unlink')
+    def test_mail_composer_delete(self):
+        """ Check mail / msg delete support """
+        # ensure initial data
+        self.assertTrue(self.template.auto_delete)
+        self.assertEqual(self.test_records.user_id, self.user_employee_2)
+        self.assertEqual(self.test_records.message_partner_ids, self.partner_employee_2)
+
+        composer_form = Form(self.env['mail.compose.message'].with_context(
+            self._get_web_context(self.test_records, add_web=True,
+                                  default_template_id=self.template.id)
+        ))
+        composer = composer_form.save()
+        with self.mock_mail_gateway(mail_unlink_sent=True), self.mock_mail_app():
+            composer._action_send_mail()
+
+        self.assertEqual(len(self._mails), 2, 'Should have sent 1 email per record')
+        self.assertEqual(len(self._new_mails), 2, 'Should have created 1 mail.mail per record')
+        self.assertEqual(self._new_mails.exists(), self.env['mail.mail'], 'Should have deleted mail.mail records')
```

Comment:
can you do this instead ?
```suggestion
        self.assertFalse(self._new_mails.exists(), 'Should have deleted mail.mail records')
```

Reply:
Yes, forgot we can do it :)